tests/integration/test_hyper_transformer.py

```diff
@@ -1370,3 +1371,636 @@ def test_random_seed(self):
+
+    # Test resetting randomization
+    ht1.reset_randomization()
+
+    transformed_post_reset = ht1.reverse_transform(first_transformed1)
+    pd.testing.assert_frame_equal(transformed_post_reset, expected_first_reverse)
+
+
+def test_hyper_transformer_cluster_based_normalizer_randomization():
+    """Test that the ``ClusterBasedNormalizer`` handles randomization correctly.
+
+    If the ``ClusterBasedNormalizer`` transforms the same data multiple times,
+    it may yield different results due to randomness. However, if a new instance is created,
+    each matching call should yield the same results (ie. call 1 of the first transformer
+    should match call 1 of the second).
+    """
+    data = get_demo(100)
+    ht = HyperTransformer()
+    ht.detect_initial_config(data)
+    ht.update_transformers({
+        'age': ClusterBasedNormalizer()
+    })
+    ht.fit(data)
+    transformed1 = ht.transform(data)
+    transformed2 = ht.transform(data)
+
+    assert any(transformed1['age.normalized'] != transformed2['age.normalized'])
+
+    ht2 = HyperTransformer()
+    ht2.detect_initial_config(data)
+    ht2.update_transformers({
+        'age': ClusterBasedNormalizer()
+    })
+    ht2.fit(data)
+
+    pd.testing.assert_frame_equal(transformed1, ht2.transform(data))
+
+
+def test_hypertransformer_anonymized_faker():
+    """Test ``AnonymizedFaker`` generates different random values for different columns.
+
+    Issue: https://github.com/sdv-dev/RDT/issues/619.
+    """
+    # Setup
+    data = pd.DataFrame({
+        'id1': ['a', 'b', 'c'],
+        'id2': ['d', 'e', 'f'],
+    })
+    ht = HyperTransformer()
+
+    # Run - simple run
+    ht.detect_initial_config(data)
+    ht.update_sdtypes({
+        'id1': 'pii',
+        'id2': 'pii'
+    })
+    ht.update_transformers({
+        'id1': AnonymizedFaker(),
+        'id2': AnonymizedFaker()
+    })
+    ht.fit(data)
+    transformed = ht.transform(data)
+    reverse_transformed1 = ht.reverse_transform(transformed)
+
+    # Assert
+    assert reverse_transformed1['id1'].tolist() != reverse_transformed1['id2'].tolist()
+
+    # Run - make sure transforming again returns different values than the original transform
+    transformed = ht.transform(data)
+    reverse_transformed2 = ht.reverse_transform(transformed)
+
+    # Assert
+    assert reverse_transformed2['id1'].tolist() != reverse_transformed2['id2'].tolist()
+    assert reverse_transformed1['id1'].tolist() != reverse_transformed2['id1'].tolist()
+    assert reverse_transformed1['id2'].tolist() != reverse_transformed2['id2'].tolist()
+
+    # Run - make sure resetting randomization works
+    ht.reset_randomization()
+    transformed = ht.transform(data)
+    reverse_transformed3 = ht.reverse_transform(transformed)
+
+    # Assert
+    pd.testing.assert_frame_equal(reverse_transformed1, reverse_transformed3)
+
+
+def test_hypertransformer_pseudo_anonymized_faker():
```

Comment:
I think some of these tests are duplicates. You may have to fix the rebase

Reply:
Yes you're right, fixed in [e8d956c](https://github.com/sdv-dev/RDT/pull/695/commits/e8d956c4db8162a28e57047d74225fe5fa3b168c)